ecosystem/typescript/sdk/src/plugins/token_client.ts

```diff
@@ -396,23 +388,15 @@ export class TokenClient {
     amount: AnyNumber,
     extraArgs?: OptionalTransactionArgs,
   ): Promise<string> {
-    // compile script to invoke public transfer function
-    const payload = new TransactionPayloadScript(
-      new Script(
-        new HexString(TOKEN_TRANSFER_OPT_IN).toUint8Array(),
-        [],
-        [
-          new TransactionArgumentAddress(TxnBuilderTypes.AccountAddress.fromHex(creator)),
-          new TransactionArgumentU8Vector(new TextEncoder().encode(collectionName)),
-          new TransactionArgumentU8Vector(new TextEncoder().encode(tokenName)),
-          new TransactionArgumentU64(BigInt(propertyVersion)),
-          new TransactionArgumentAddress(TxnBuilderTypes.AccountAddress.fromHex(receiver)),
-          new TransactionArgumentU64(BigInt(amount)),
-        ],
-      ),
+    const builder = new TransactionBuilderRemoteABI(this.aptosClient, { sender: sender.address(), ...extraArgs });
+    const rawTxn = await builder.build(
+      "0x3::token::transfer_with_opt_in", 
+      [], 
```

Comment:
I assume there is no _ty_tags_ needed here. But I am curious when is this needed?

Reply:
`type_arguments` needed when the move function expects typed arguments, i.e https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-token/sources/token_coin_swap.move#L103

if it doesn't we simply pass in an empty array - though we should probably be more sophisticate than that as with the remote abi we should know if the function expects it, and if it doesn't we should be able to not even ask for it, but that is a different discussion not related to this PR